linker/shared/src/test/scala/org/scalajs/linker/LinkerTest.scala

```diff
@@ -50,6 +52,39 @@ class LinkerTest {
     testLink(helloWorldClassDefs, MainTestModuleInitializers)
   }
 
+  /** Makes sure that linking a "substantial" program (using `println`) does
+   *  not trigger the second attempt in the Emitter due to dangerous global
+   *  refs.
+   */
+  @Test
+  def linkNoSecondAttemptInEmitter(): AsyncResult = await {
```

Comment:
Put this in a new class `EmitterTest`? Even though it goes through the interface of the linker, it tests the emitter (we have a lot of precedent in tests here to do that).